### PR TITLE
TabView Visual Updates

### PR DESCRIPTION
**Pull request by beervoley**
## Description

This PR updates the current TabView to use new colors and updates some TabView logic.
## Motivation and Context

TabView mostly relied on shadows to draw the outlines of the Tabs but the new design proposed using strokes to make it look a little bit cleaner.
Some of the changes:
1) Arcs paths were modified a little to add a stroke.
2) All the buttons were visually updated.
3) When hovering over / pressing anon-selected tab both the tab separators (on hovered over /pressed tab as well as on the tab adjacent on the left) will disappear.
## How Has This Been Tested?

Manually.
## Screenshots (if appropriate):
Light:

image
image
image
image
image

Dark:
image
image
image
image
image

Disappearing tab separators:
scrollviewer_bug

### CODE REVIEW

**Comment by beervoley** [comment]
@chingucoding could you please have a look?:)
I've also removed code for the shadows but I think we still need it for the old style to function properly.

File: dev/TabView/TabView.cpp [hunk 1]
```diff
@@ -590,14 +606,21 @@ void TabView::OnItemsChanged(winrt::IInspectable const& item)
         m_tabItemsChangedEventSource(*this, args);
 
         int numItems = static_cast<int>(TabItems().Size());
+        const auto listViewInnerSelectedIndex = m_listView.get().SelectedIndex();
+        auto selectedIndex = SelectedIndex();
+        
+        if (selectedIndex != listViewInnerSelectedIndex && listViewInnerSelectedIndex != -1)
+        {
+            SelectedIndex(listViewInnerSelectedIndex);
+            selectedIndex = SelectedIndex();
```
Comment (StephenLPeters) [inline dev/TabView/TabView.cpp:615]: >SelectedIndex [](http://example.com/codeflow?start=28&length=13)

nit: selectedIndex = listViewInnerSelectedIndex;

avoid using the DP getter if you don't need to.

File: dev/TabView/TabView.xaml [hunk 2]
```diff
@@ -683,23 +709,87 @@
...
+        <Setter Property="Background" Value="{ThemeResource TabViewButtonBackground}"/>
+        <Setter Property="Foreground" Value="{ThemeResource TabViewButtonForeground}"/>
+        <contract7Present:Setter Property="CornerRadius" Value="{ThemeResource ControlCornerRadius}"/>
+        <Setter Property="FontSize" Value="{ThemeResource TabViewItemScrollButonFontSize}"/>
+        <Setter Property="FontFamily" Value="{ThemeResource SymbolThemeFontFamily}"/>
+        <Setter Property="Width" Value="{ThemeResource TabViewItemScrollButtonWidth}"/>
+        <Setter Property="Height" Value="{ThemeResource TabViewItemScrollButtonHeight}"/>
+        <Setter Property="FontFamily" Value="{ThemeResource SymbolThemeFontFamily}"/>
+        <Setter Property="FocusVisualMargin" Value="-3"/>
+        <Setter Property="Template">
```
Comment (StephenLPeters) [inline dev/TabView/TabView.xaml:735]: >Template [](http://example.com/codeflow?start=26&length=8)

What is the new template needed for?

File: dev/TabView/TabView.cpp (hunk 1 above)
Comment (beervoley) [reply dev/TabView/TabView.cpp:615]: will do

File: dev/TabView/TabView.xaml (hunk 2 above)
Comment (beervoley) [reply dev/TabView/TabView.xaml:735]: for correct foreground/background colors in visual states:)

File: dev/TabView/TabView.xaml [hunk 3]
```diff
@@ -683,23 +709,87 @@
...
+            <Setter.Value>
+                <ControlTemplate TargetType="RepeatButton">
+                    <ContentPresenter x:Name="ContentPresenter"
+                        Background="{TemplateBinding Background}"
+                        contract7Present:BackgroundSizing="{TemplateBinding BackgroundSizing}"
+                        Content="{TemplateBinding Content}"
+                        ContentTemplate="{TemplateBinding ContentTemplate}"
+                        ContentTransitions="{TemplateBinding ContentTransitions}"
+                        contract7Present:CornerRadius="{TemplateBinding CornerRadius}"
+                        contract7NotPresent:CornerRadius="{ThemeResource TabViewItemButtonCornerRadius}"
```
Comment (StephenLPeters) [inline dev/TabView/TabView.xaml]: I don't see TabViewItemButtonCornerRadius defined anywhere..

Comment (beervoley) [reply dev/TabView/TabView.xaml]: fixed:)

File: dev/TabView/TabView.xaml [hunk 4]
```diff
@@ -683,23 +709,87 @@
...
+        <Setter Property="FontFamily" Value="{ThemeResource SymbolThemeFontFamily}"/>
+        <Setter Property="Width" Value="{ThemeResource TabViewItemScrollButtonWidth}"/>
+        <Setter Property="Height" Value="{ThemeResource TabViewItemScrollButtonHeight}"/>
+        <Setter Property="FontFamily" Value="{ThemeResource SymbolThemeFontFamily}"/>
+        <Setter Property="FocusVisualMargin" Value="-3"/>
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate TargetType="RepeatButton">
+                    <ContentPresenter x:Name="ContentPresenter"
+                        Background="{TemplateBinding Background}"
```
Comment (StephenLPeters) [inline dev/TabView/TabView.xaml:739]: It looks like you deleted the setters for BorderBrush and BorderThickness.  I thikn better would be to set those properties to the correct values.

File: dev/TabView/TabView.xaml [hunk 5]
```diff
@@ -683,23 +709,87 @@
...
+                        contract7Present:BackgroundSizing="{TemplateBinding BackgroundSizing}"
+                        Content="{TemplateBinding Content}"
+                        ContentTemplate="{TemplateBinding ContentTemplate}"
+                        ContentTransitions="{TemplateBinding ContentTransitions}"
+                        contract7Present:CornerRadius="{TemplateBinding CornerRadius}"
+                        contract7NotPresent:CornerRadius="{ThemeResource TabViewItemButtonCornerRadius}"
+                        FontSize="{TemplateBinding FontSize}"
```
Comment (StephenLPeters) [inline dev/TabView/TabView.xaml:746]: These font properties are propogating, so setting them explicitly shouldn't be necissary.

File: dev/TabView/TabView.xaml (hunk 2 above)
Comment (StephenLPeters) [reply dev/TabView/TabView.xaml:735]: We should try not to retemplate, it makes our lives harder in a lot of ways.  That being said, it looks like you removed the default template's BackgroundTransition. I think the only way to do that is via a retemplate, so I think this is alright.

**Review by StephenLPeters — APPROVED** [review]
:shipit:

**Pull request merged**